File: Walmart.Sgp.WebApp/Scripts/app/services/alcadaDetalhe.service.js
(function () {
    'use strict';

    angular
        .module('SGP')
        .service('AlcadaDetalheService', AlcadaDetalheService);

    AlcadaDetalheService.$inject = ['$q', '$http', 'ApiEndpoints'];

    function AlcadaDetalheService($q, $http, ApiEndpoints) {

        function retornarDados(response) {
            return response.data;
        }

        var self = this;

        self.obterPorIdAlcada = function (idAlcada, paging) {
            var params = ApiEndpoints.createParams(paging, {
                idAlcada: idAlcada
            });

            return $http
                .get(ApiEndpoints.sgp.alcadaDetalhe, { params: params })
                .then(retornarDados);
        };
    }
})();

File: Walmart.Sgp.Domain/Acessos/AlcadaDetalheService.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Walmart.Sgp.Infrastructure.Framework.Domain;
using Walmart.Sgp.Infrastructure.Framework.Specs;

namespace Walmart.Sgp.Domain.Acessos
{
    /// <summary>
    /// Serviço de domínio relacionado a AlcadaDetalhe.
    /// </summary>
    public class AlcadaDetalheService : EntityDomainServiceBase<AlcadaDetalhe, IAlcadaDetalheGateway>, IAlcadaDetalheService
    {
        /// <summary>
        /// Inicia uma nova instância da classe <see cref="AlcadaDetalheService"/>.
        /// </summary>
        /// <param name="mainGateway">O table data gateway para main data.</param>
        public AlcadaDetalheService(IAlcadaDetalheGateway mainGateway)
            : base(mainGateway)
        {
        }

        /// <summary>
        /// Obter uma lista de AlcadaDetalhe por IdAlcada.
        /// </summary>
        /// <param name="idAlcada">O id da Alcada.</param>
        /// <param name="paging">A paginação.</param>
        /// <returns>A lista de AlcadaDetalhe.</returns>
        public IEnumerable<AlcadaDetalhe> ObterPorIdAlcada(int idAlcada, Paging paging)
        {
            Assert(new { Competence = idAlcada }, new AllMustBeInformedSpec());

            return this.MainGateway.ObterPorIdAlcada(idAlcada, paging);
        }
    }
}
